featuregate/registry.go

```diff
@@ -28,43 +38,61 @@ func NewRegistry() *Registry {
 
 // RegisterOption allows to configure additional information about a Gate during registration.
 type RegisterOption interface {
-	apply(g *Gate)
+	apply(g *Gate) error
 }
 
-type registerOptionFunc func(g *Gate)
+type registerOptionFunc func(g *Gate) error
 
-func (ro registerOptionFunc) apply(g *Gate) {
-	ro(g)
+func (ro registerOptionFunc) apply(g *Gate) error {
+	return ro(g)
 }
 
 // WithRegisterDescription adds description for the Gate.
 func WithRegisterDescription(description string) RegisterOption {
-	return registerOptionFunc(func(g *Gate) {
+	return registerOptionFunc(func(g *Gate) error {
 		g.description = description
+		return nil
 	})
 }
 
 // WithRegisterReferenceURL adds a URL that has all the contextual information about the Gate.
-func WithRegisterReferenceURL(url string) RegisterOption {
-	return registerOptionFunc(func(g *Gate) {
-		g.referenceURL = url
+func WithRegisterReferenceURL(referenceURL string) RegisterOption {
+	return registerOptionFunc(func(g *Gate) error {
+		if _, err := url.Parse(referenceURL); err != nil {
+			return fmt.Errorf("WithRegisterReferenceURL: invalid reference URL %q: %w", referenceURL, err)
+		}
+
+		g.referenceURL = referenceURL
+		return nil
 	})
 }
 
 // WithRegisterFromVersion is used to set the Gate "FromVersion".
 // The "FromVersion" contains the Collector release when a feature is introduced.
 func WithRegisterFromVersion(fromVersion string) RegisterOption {
-	return registerOptionFunc(func(g *Gate) {
-		g.fromVersion = fromVersion
+	return registerOptionFunc(func(g *Gate) error {
+		from, err := version.NewVersion(fromVersion)
+		if err != nil {
+			return fmt.Errorf("WithRegisterFromVersion: %w", err)
+		}
+
+		g.fromVersion = from
+		return nil
 	})
 }
 
 // WithRegisterToVersion is used to set the Gate "ToVersion".
 // The "ToVersion", if not empty, contains the last Collector release in which you can still use a feature gate.
 // If the feature stage is either "Deprecated" or "Stable", the "ToVersion" is the Collector release when the feature is removed.
 func WithRegisterToVersion(toVersion string) RegisterOption {
```

Comment:
Please explain in the comment what is a valid version.

Reply:
Done